package lab1_exceptions;

public class ExceptionsLab1 {

	public static void main(String[] args) {
		
		testMethodA("a");
		System.out.println();
		testMethodB();
		System.out.println();
		testMethodC();

	}


public static void testMethodA(String a){
	try{
		System.out.println("Method A");
	}
	catch (Exception e){
		System.out.println("Catch Exception" + e);
	}
	finally{
		System.out.println("OK from finally");
	}
}

public static String testMethodB(){
	try{
		System.out.println("Method B");
	}
	catch (Exception e){
		System.out.println("Exception: " + e);
	}
	finally{
		System.out.println("Finally section");
	}
	return null;
}

public static String testMethodC(){
	try{
		return "Method C";
	}
	catch (Exception e){
		return "Exception: " + e;
	}
	finally {
		return "Finally";
	}
}


public static String testMethodD(){
	try{
		return "Method D";
	}
	catch (Exception e){
		return "Exception: " + e;
	}
	finally{
		return "Finally";
	}
}
}
